app/javascript/components/record-form/form/subforms/subform-fields/components/violation-item/component.jsx

```diff
@@ -1,19 +1,130 @@
+import { toServerDateFormat } from "../../../../../../../libs";
 
-import { NAME } from "./constants";
+import VerifySelect from "./select";
 import { getViolationTallyLabel } from "./utils";
+import { NAME } from "./constants";
+
+import { KeyboardDatePicker, MuiPickersUtilsProvider } from "@material-ui/pickers";
+import DateFnsUtils from "@date-io/date-fns";
+
 
-const Component = ({ fields, values, locale, displayName, index, collapsedFieldValues }) => {
+const Component = ({ fields, values, locale, displayName, index, collapsedFieldValues, mode }) => {
   const currentValues = values[index];
+  const verifyParams = useParams();
+  const i18n = useI18n();
+  const DATE_FORMAT = "dd-MMM-yyyy";
+
+  // State variables
+  const dispatch = useDispatch();
+  const [verifyModal, setVerifyModal] = useState(false);
+  const [verificationValue, setVerificationValue] = useState(currentValues?.ctfmr_verified || ""); 
+  const [selectedDate, setSelectedDate] = useState(new Date());
+  const [validationError, setValidationError] = useState("");
+
+  useEffect(() => {
+    // Changing dropdown select value when backend data updated
+    setVerificationValue(currentValues?.ctfmr_verified);
+  }, [currentValues?.ctfmr_verified]);
+
+  const handleDropdownDate = (date) => {
+    if (selectedDate) {
+      setSelectedDate(date);
+    } else {
+      setValidationError("date should not be null");
+      setSelectedDate(null);
+    }
+  };
+
+  const handleOpenVerifyModal = (index, event) => {
+    //  To open verify dialog confirmation popup
+    event.stopPropagation();
+    setVerifyModal(true);
+  };
 
+  const cancelVerifyHandler = () => {
+    //  To close verify dialog popup
+    setVerifyModal(false);
+  };
+
+  const canVerify = usePermissions("incidents", RECORD_ACTION_ABILITIES); //check permission to verify violations
   const violationTally = getViolationTallyLabel(fields, currentValues, locale);
 
+  const handleOk = () => {
+    //  To update the verify status to Verified
+
+    dispatch(
+      saveRecord(
+        verifyParams.recordType,
+        "update",
+        {
+          data: {
+            [currentValues.type]: [
+              {
+                unique_id: currentValues.unique_id,
+                ctfmr_verified: verificationValue,
+                ctfmr_verified_date: toServerDateFormat(selectedDate)
+              }
+            ]
+          }
+        }, // Save API Call
+        verifyParams.id,
+        "Updated successfully",
+        "",
+        false,
+        false
+      )
+    );
+    // close();
+  };
+
+  // Define VerifySelect component
+  const VerifySelectComponent = (
+    <VerifySelect
+      selectedValue={verificationValue}
+      setSelectedValue={setVerificationValue}
+    />
+  );
+
+  // Define MuiPickersUtilsProvider component
+  const MuiPickersUtilsProviderComponent = (
+    <MuiPickersUtilsProvider utils={DateFnsUtils}>
+      <div style={{ display: "flex", justifyContent: "space-between" }}>
```

Comment:
Still have some props that are objects. Styles like this can be move into a css file and referenced with the className prop.